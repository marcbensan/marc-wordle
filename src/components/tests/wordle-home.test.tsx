import WordleHome from "@/components/wordle-home";
import { cleanup, fireEvent, render, screen } from "@testing-library/react";
import { beforeEach, expect, test, vi } from "vitest";

const mockPush = vi.fn();

vi.mock("next/navigation", () => ({
  useRouter: () => ({
    push: mockPush,
  }),
}));

beforeEach(() => {
  cleanup();
});

test("Match WordleHome snapshot", () => {
  expect(render(<WordleHome />)).toMatchSnapshot();
});

test("Renders the WordleHome component correctly", () => {
  render(<WordleHome />);

  expect(screen.getByText(/Enjoy a quick game of/i)).toBeDefined();
  expect(screen.getByText("WORDLE")).toBeDefined();
  expect(screen.getByText("PLAY")).toBeDefined();
});

test("Redirects to game page when PLAY button is clicked", () => {
  render(<WordleHome />);
  fireEvent.click(screen.getByText("PLAY"));

  expect(mockPush).toHaveBeenCalledWith("/game");
});
