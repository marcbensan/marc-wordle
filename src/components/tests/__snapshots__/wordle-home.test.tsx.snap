// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Match WordleHome snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col overflow-hidden mb-48"
      >
        <div
          class="h-[60rem] md:h-[70rem] flex items-center justify-center relative p-2 md:p-20"
        >
          <div
            class="py-10 md:py-40 w-full relative"
            style="perspective: 1000px;"
          >
            <div
              class="div max-w-5xl mx-auto text-center"
              style="transform: none;"
            >
              <h1
                class="text-4xl font-semibold text-white"
              >
                Enjoy a quick game of 
                <br />
                <span
                  class="text-4xl md:text-[6rem] font-bold mt-1 leading-none"
                >
                  WORDLE
                </span>
              </h1>
            </div>
            <div
              class="max-w-5xl -mt-12 mx-auto h-[30rem] md:h-[40rem] w-full border-4 border-[#6C6C6C] p-2 md:p-6 bg-[#222222] rounded-[30px] shadow-2xl"
              style="box-shadow: 0 0 #0000004d, 0 9px 20px #0000004a, 0 37px 37px #00000042, 0 84px 50px #00000026, 0 149px 60px #0000000a, 0 233px 65px #00000003; transform: scale(1.05) rotateX(20deg);"
            >
              <div
                class=" h-full w-full  overflow-hidden rounded-2xl bg-gray-100 dark:bg-zinc-900 md:rounded-2xl md:p-4 "
              >
                <img
                  alt="hero"
                  class="mx-auto rounded-2xl object-cover h-full "
                  data-nimg="1"
                  decoding="async"
                  draggable="false"
                  height="720"
                  loading="lazy"
                  src="/_next/image?url=%2Fimages%2Fwordle-pic.png&w=3840&q=75"
                  srcset="/_next/image?url=%2Fimages%2Fwordle-pic.png&w=1920&q=75 1x, /_next/image?url=%2Fimages%2Fwordle-pic.png&w=3840&q=75 2x"
                  style="color: transparent;"
                  width="1400"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="w-full flex items-center font-subheading justify-center"
        >
          <button
            class="items-center justify-center gap-2 whitespace-nowrap font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive shadow-xs px-4 py-2 has-[>svg]:px-3 text-4xl cursor-pointer text-wordle-dark hover:bg-wordle-light/80 flex space-between bg-wordle-light h-18 w-56 rounded-full"
            data-slot="button"
          >
            PLAY
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col overflow-hidden mb-48"
    >
      <div
        class="h-[60rem] md:h-[70rem] flex items-center justify-center relative p-2 md:p-20"
      >
        <div
          class="py-10 md:py-40 w-full relative"
          style="perspective: 1000px;"
        >
          <div
            class="div max-w-5xl mx-auto text-center"
            style="transform: none;"
          >
            <h1
              class="text-4xl font-semibold text-white"
            >
              Enjoy a quick game of 
              <br />
              <span
                class="text-4xl md:text-[6rem] font-bold mt-1 leading-none"
              >
                WORDLE
              </span>
            </h1>
          </div>
          <div
            class="max-w-5xl -mt-12 mx-auto h-[30rem] md:h-[40rem] w-full border-4 border-[#6C6C6C] p-2 md:p-6 bg-[#222222] rounded-[30px] shadow-2xl"
            style="box-shadow: 0 0 #0000004d, 0 9px 20px #0000004a, 0 37px 37px #00000042, 0 84px 50px #00000026, 0 149px 60px #0000000a, 0 233px 65px #00000003; transform: scale(1.05) rotateX(20deg);"
          >
            <div
              class=" h-full w-full  overflow-hidden rounded-2xl bg-gray-100 dark:bg-zinc-900 md:rounded-2xl md:p-4 "
            >
              <img
                alt="hero"
                class="mx-auto rounded-2xl object-cover h-full "
                data-nimg="1"
                decoding="async"
                draggable="false"
                height="720"
                loading="lazy"
                src="/_next/image?url=%2Fimages%2Fwordle-pic.png&w=3840&q=75"
                srcset="/_next/image?url=%2Fimages%2Fwordle-pic.png&w=1920&q=75 1x, /_next/image?url=%2Fimages%2Fwordle-pic.png&w=3840&q=75 2x"
                style="color: transparent;"
                width="1400"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="w-full flex items-center font-subheading justify-center"
      >
        <button
          class="items-center justify-center gap-2 whitespace-nowrap font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive shadow-xs px-4 py-2 has-[>svg]:px-3 text-4xl cursor-pointer text-wordle-dark hover:bg-wordle-light/80 flex space-between bg-wordle-light h-18 w-56 rounded-full"
          data-slot="button"
        >
          PLAY
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
